packages/apollo-server-core/src/requestPipelineAPI.ts

```diff
@@ -10,3 +23,143 @@ export {
   GraphQLExecutor,
   GraphQLExecutionResult,
 } from 'apollo-server-types';
+
+export const symbolRequestListenerDispatcher =
+  Symbol("apolloServerRequestListenerDispatcher");
+export const symbolPluginsEnabled = Symbol("apolloServerPluginsEnabled");
+
+export function enablePluginsForSchemaResolvers(
+  schema: GraphQLSchema & { [symbolPluginsEnabled]?: boolean },
+) {
+  if (schema[symbolPluginsEnabled]) {
+    return schema;
+  }
+  Object.defineProperty(schema, symbolPluginsEnabled, {
+    value: true,
+  });
+
+  forEachField(schema, wrapField);
+
+  return schema;
+}
+
+function wrapField(field: GraphQLField<any, any>): void {
+  const fieldResolver = field.resolve || defaultFieldResolver;
```

Comment:
The only real change from the extensions implementation (other than the use of Symbols, which is cool) is dropping the "runtime" check for a `this.fieldResolver` that comes in between the field's own resolver and the defaultFieldResolver. While there was a comment from Martijn claiming that this.fieldResolver is never set, I don't think that's actually true — understanding apollo-server configuration is complex but I *believe* that `fieldResolver` is part of that `GraphQLServerOptions` that can be configured on a per-request basis, which gets passed to the extension stack's executionDidStart. So I think we're actually losing a feature of apollo-server by not plumbing this user-provided fieldResolver down to the wrapped fields.  Perhaps define a third context symbol that can save the fieldResolver on the context?

In related news, there's a comment in executeQueryPlan.ts:
```
      // FIXME: GraphQL extensions currently wraps every field and creates
      // a field resolver. Because of this, when using with ApolloServer
      // the defaultFieldResolver isn't called. We keep this here
      // because it is the correct solution and when ApolloServer removes
      // GraphQLExtensions this will be how alias support is maintained
      fieldResolver: defaultFieldResolverWithAliasSupport,
```
I believe what it's saying is that passing a fieldResolver is basically dead code because wrapField means the fieldResolver passed to `graphql-js`' `execute` directly (what is being done over there) is ignored.  Probably update this comment (unless you did that in another PR?) to no longer hint at an intention to change how wrapField works even though we are removing GraphQLExtensions?

Reply:
@glasser I agree that tracking this is more nuanced and tricky than ideal.  Does this commit — which introduces a test-case in the context of plugin hooks and I've pushed to this PR's branch — help make this any more clear?: 60cd32d0cc35b02fcdf57805dd450f4f07cfb55f

I want to say that this still works because of `graphql/execution` [logic in `buildExecutionContext`](https://github.com/graphql/graphql-js/blob/bb104d9760b255689745aaa00bd49d8a0500a11e/src/execution/execute.js#L335) combined with [this logic in `resolveField`](https://github.com/graphql/graphql-js/blob/bb104d9760b255689745aaa00bd49d8a0500a11e/src/execution/execute.js#L603-L618) in `graphql`'s `execute.js`?  Thus, the default resolve _**or a custom-provided `fieldResolver`** via our own `GraphQLServerOptions`_ is made available when `wrapField` is invoked?

Help to confirm? (Or let my test help?)

(I'm going to zoom in more on the `executeQueryPlan` bit you highlighted above for now.)

Reply:
I don't understand how your test passes. Will try to figure that out!

Reply:
The test passes because you are still installing graphql-extensions on the schema, and you do that before installing plugins, so the plugin wrapper wraps the graphql-extensions wrapper which still implements fieldResolver. Your test fails if you comment enableGraphQLExtensions() :)

Reply:
I'm glad we have a test for it now which will fall down when we actually remove GraphQL Extensions.

We will need to fix this, however I think it will be easier to feel comfortable about the solution after we delete the GraphQL extensions.  The user's `fieldResolver` is being preserved right now, and we have a test to ensure that it is preserved in the future.  It will absolutely break when we delete `GraphQLExtension`s, and I think introducing the `fieldResolver` (or `defaultFieldResolveWithAliasSupport`) to the `context` on a symbol in a same/similar way to the `willResolveField` dispatcher will be a way to resolve it.

The 3.x work will kick off soon, and getting that done is largely dependent on landing this PR along with [the multitude of PRs that merge into it](https://github.com/apollographql/apollo-server/milestone/43).

Does that sound like an okay / safe plan?

Reply:
Ok, I second guessed my comfort with leveraging this and took a shot at the solution with 6564081240ee884f3d6c5522ed17f1b48.  😸 

It's seeming to work with related PRs with the `wrapField` of `graphql-extensions` commented out, e.g. the cache-control refactor in https://github.com/apollographql/apollo-server/pull/3997 and the AER refactor in https://github.com/apollographql/apollo-server/pull/3998.

Reply:
I hear your concern about the fact that the new code is undertested. I think it's pretty likely that it works given the manual testing though (and that it's not super complex).

Reply:
Re the comment in executeQueryPlan do you want to at least update it to not imply that "remove graphql-extensions" implies "significant changes to how we wrap fields"? Honestly I think the whole comment is maybe removable. My current understanding (distinct from what I was thinking a few weeks ago) is that that line is absolutely necessary to process aliases correctly, and there's no need for an extended FIXME comment relating to graphql-extensions at all.

Reply:
I just found another comment when looking for `defaultFieldResolverWithAliasSupport`:

https://github.com/apollographql/apollo-server/blob/52418a4f3732b004f581655d426d20df21c9e0be/packages/apollo-gateway/src/index.ts#L479-L483

Anyhow, I've made some changes: 18d95fd7f30bba196f7b0b62abdcff6bcb0eae18

I will certainly double-click on this when we remove `graphql-extensions` in the future, especially now that the surface area of concerns is greatly reduced.

Thanks for bringing this up originally.  I've learned a lot.